app/routers/quizzes.py

```diff
@@ -92,6 +93,35 @@ async def get_quiz(quiz_id: str):
     quiz_collection = client.quiz.quizzes
     if (quiz := quiz_collection.find_one({"_id": quiz_id})) is not None:
         update_quiz_for_backwards_compatibility(quiz_collection, quiz_id, quiz)
+
+        if quiz["metadata"]["quiz_type"] == QuizType.omr.value:
+            print("true", QuizType.omr)
+            for question_set_index, question_set in enumerate(quiz["question_sets"]):
+                options_count_aggregate = client.quiz.questions.aggregate(
+                    [
+                        {"$match": {"question_set_id": question_set["_id"]}},
+                        {"$sort": {"_id": 1}},
+                        {"$skip": settings.subset_size},
+                        {"$project": {"number_of_options": {"$size": "$options"}}},
+                    ]
+                )
+                options_count = [
+                    doc["number_of_options"] for doc in list(options_count_aggregate)
+                ]
```

Comment:
This is not at all necessary. This is an extra loop over all the values where you're just extracting `number_of_options` from the array of objects. You can add two stages to your aggregation query to return you the array of number directly instead of array of objects.
This is how the final query might look like
```
[
  {
      '$match': {
          'question_set_id': '64344359edcb8c78aef1ce1e'
      }
  }, {
      '$sort': {
          '_id': 1
      }
  }, {
      '$skip': 10
  }, {
      '$project': {
          '_id': 0, 
          'number_of_options': {
              '$size': '$options'
          }
      }
  }, {
      '$group': {
          '_id': None, 
          'optionsArray': {
              '$push': '$number_of_options'
          }
      }
  }, {
      '$project': {
          '_id': 0, 
          'optionsArray': 1
      }
  }
]
```

(chat gpt helped me with this :p)